elementpath/xpath1/_xpath1_operators.py

```diff
@@ -371,7 +364,7 @@ def select_self_shortcut(self, context=None):
 
     elif isinstance(context, XPathSchemaContext):
         for item in context.iter_self():
-            if is_schema_node(item):
+            if item.is_schema_element():
```

Comment:
Because ElementNode instances are used for wrapping also schema nodes and XSD elements (for xmlschema XPath selection and for schema type labelling in XP20+).